aria/model/model.py

```diff
@@ -31,35 +31,45 @@ def set_vocab_size(self, vocab_size: int):
 
 
 # Taken from GPT-NeoX see:
-# https://github.com/EleutherAI/gpt-neox/blob/main/megatron/model/positional_embeddings.py
+# https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt_neox/modeling_gpt_neox.py
 class RotaryEmbedding(torch.nn.Module):
-    def __init__(self, dim, base=10000, precision=torch.half):
+    def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None):
         super().__init__()
-        inv_freq = 1.0 / (base ** (torch.arange(0, dim, 2).float() / dim))
+        if device is None: # todo: maybe we don't need this...
```

Comment:
Probably best to not explicitly do this. Could instead perhaps do `device = self.device`? I'm worried that this will break the distributed code (with accelerate) but I'm unable to test it at the moment.

Reply:
The problem with device=self.device is that somehow device=None defaults to CPU and I will get mismatch device error.

Reply:
@loubbrad I didn't realize these two lines can be safely removed now with hugging face codes. In fact `device` param itself is redundant!
In the original implementation of RotaryEmbedding (from neox repo) this device needs to be known at initialization-time because the cached `cos_cached, sin_cached` need to be explicitly set. But now they are registered as model parameters and moving across device is automatic.